numpyro/hmc_util.py

```diff
@@ -578,9 +578,6 @@ def _body_fn(state):
         return tree, key
 
     state = (tree, key)
-    if iterative_build:
-        tree, _ = lax.while_loop(_cond_fn, _body_fn, state)
-    else:
-        while _cond_fn(state):
-            tree, _ = state = _body_fn(state)
+    with optional(not iterative_build, control_flow_prims_disabled()):
```

Comment:
Note that this isn't equivalent to the version on the left, because it sets the `_DISABLE_CONTROL_FLOW_PRIM` flag for other operations inside `build_tree` which won't work unless jit is disabled as well. I will revert this change.